File: Java/Problem2.java
class Solution {
    public int minPatches(int[] nums, int n) {
        int resultado = 0;
        int faltantes = 1;
        int ina = 0;

        while (faltantes <= n) {
            if (ina < nums.length && nums[ina] <= faltantes){
                faltantes += nums[ina];
                ina++;
            } 
            else {
                faltantes += faltantes;
                resultado++;
            }
        }
        return resultado;
    }
}

File: Python/Problem2.py
class Solution(object):
    def minPatches(self, nums, n):
        faltantes = 1
        resultado = 0
        ina = 0

        while faltantes <= n:
            if ina < len(nums) and nums[ina] <= faltantes:
                faltantes += nums[ina]
                ina += 1
            else:
                faltantes += faltantes
                resultado += 1
        return resultado
